### ISSUE TITLE
Can I download and play network audio at the same time?

### ISSUE THREAD

**Issue by liutian0117** [tanhakabir/SwiftAudioPlayer#101]

**Comment by tanhakabir** [comment]
You can download a different audio and stream at the same time. But unfortunately you can't download and stream the same audio at the same time currently.

**Comment by liutian0117** [comment]
Why is this happening?
Is it because dataTask and downloadTask cannot be executed on one URL at the same time?

**Comment by liutian0117** [comment]
> You can download a different audio and stream at the same time. But unfortunately you can't download and stream the same audio at the same time currently.

Why is this happening?
Is it because dataTask and downloadTask cannot be executed on one URL at the same time?

**Comment by tanhakabir** [comment]
@liutian0117 actually you could start a background download and stream the same file but it'll download them separately at the same time. I don't have the feature to save the data that's being streamed to be able to be played later from the device.

**Comment by yeralin** [comment]
@liutian0117 I tried accomplishing something similar in my player https://github.com/yeralin/OpenPlayer/blob/master/MusicPlayer/Audio/AudioDownload/RemotePlayerItem.swift

Generally it works as I use `dataTask` and feed my player with incoming data (which is then I save locally).

The main problem is that `dataTask`s are killed by OS if an app goes into background.

I started looking into `SwifttAudioPlayer` and realized they have the same problem.
